src/go/k8s/apis/redpanda/v1alpha1/cluster_types_test.go

```diff
@@ -0,0 +1,116 @@
+	type test struct {
+		name                string
+		setRequestsCPU      resource.Quantity
+		setRequestsMem      resource.Quantity
+		setRedpandaCPU      resource.Quantity
+		setRedpandaMem      resource.Quantity
+		expectedRedpandaCPU resource.Quantity
+		expectedRedpandaMem resource.Quantity
+	}
+	makeResources := func(t test) v1alpha1.RedpandaResourceRequirements {
+		return v1alpha1.RedpandaResourceRequirements{
+			ResourceRequirements: corev1.ResourceRequirements{
+				Requests: corev1.ResourceList{
+					corev1.ResourceMemory: t.setRequestsMem,
+					corev1.ResourceCPU:    t.setRequestsCPU,
+				}},
+			Redpanda: corev1.ResourceList{
+				corev1.ResourceMemory: t.setRedpandaMem,
+				corev1.ResourceCPU:    t.setRedpandaCPU,
+			},
+		}
+	}
+
+	t.Run("Memory", func(t *testing.T) {
+		tests := []test{
+			{
+				name:                "RedpandaMemory is set from requests.memory",
+				setRequestsMem:      resource.MustParse("3000Mi"),
+				expectedRedpandaMem: resource.MustParse("3000Mi"),
+			},
+			{
+				name:                "RedpandaMemory is set from lower redpanda.memory",
+				setRequestsMem:      resource.MustParse("4000Mi"),
+				setRedpandaMem:      resource.MustParse("3000Mi"),
+				expectedRedpandaMem: resource.MustParse("3000Mi"),
+			},
+		}
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				rrr := makeResources(tt)
+				assert.Equal(t, tt.expectedRedpandaMem.Value(), rrr.RedpandaMemory().Value())
+			})
+		}
+	})
+
+	t.Run("Cpu", func(t *testing.T) {
+		tests := []test{
+			{
+				name:                "RedpandaCPU is set from integer requests.cpu",
+				setRequestsCPU:      resource.MustParse("1"),
+				setRequestsMem:      resource.MustParse("20Gi"),
+				expectedRedpandaCPU: resource.MustParse("1"),
+			},
+			{
+				name:                "RedpandaCPU is set from milli requests.cpu",
+				setRequestsCPU:      resource.MustParse("1000m"),
+				setRequestsMem:      resource.MustParse("20Gi"),
+				expectedRedpandaCPU: resource.MustParse("1"),
+			},
+			{
+				name:                "RedpandaCPU is rounded up from milli requests.cpu",
+				setRequestsCPU:      resource.MustParse("1001m"),
+				setRequestsMem:      resource.MustParse("20Gi"),
+				expectedRedpandaCPU: resource.MustParse("2"),
+			},
+			{
+				name:                "RedpandaCPU is set from lower redpanda.cpu",
+				setRequestsCPU:      resource.MustParse("2"),
+				setRequestsMem:      resource.MustParse("20Gi"),
+				setRedpandaCPU:      resource.MustParse("1"),
+				expectedRedpandaCPU: resource.MustParse("1"),
+			},
+			{
+				name:                "RedpandaCPU is set from higher redpanda.cpu",
+				setRequestsCPU:      resource.MustParse("1"),
+				setRequestsMem:      resource.MustParse("20Gi"),
+				setRedpandaCPU:      resource.MustParse("2"),
+				expectedRedpandaCPU: resource.MustParse("2"),
+			},
+			{
+				name:                "RedpandaCPU is rounded up from milli redpanda.cpu",
+				setRequestsCPU:      resource.MustParse("1"),
```

Comment:
ditto

Reply:
`--smp` takes integers. `Quantity.Value()` returns a rounded up integer. It's the same behaviour as before, but moved in `RedpandaResourceRequirements` and tested,